services/repository/fork.go

```diff
@@ -134,9 +139,15 @@ func ForkRepository(ctx context.Context, doer, owner *user_model.User, opts Fork
 
 		needsRollback = true
 
+		var cloneCmd *git.Command = nil
+		if opts.SingleBranch != "" {
+			cloneCmd = git.NewCommand(txCtx, "clone", "--bare", "--single-branch", "--branch")
```

Comment:
For some reason the cloned repos keep original size whatsoever

Reply:
However cloning from the Gitea server results in fork being less than original, which is correct. FS size of the fork is approximately equal to the size of the original

Reply:
Thank you for this feature.
I think that being also able to select a tag instead of a branch could be very useful.

Reply:
You just have to provide the tag name to the `--branch` option.
From [git clone](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---branchltnamegt) documentation;
**`--branch` can also take tags and detaches the HEAD at that commit in the resulting repository.**

Reply:
Ah, alright, thanks. I'm unsure whether tags should be shown in the same selection box by default

Reply:
Maybe we can change it to

```go
cloneCmd := git.NewCommand(txCtx, "clone", "--bare")
if opts.SingleBranch != "" {
    cloneCmd.AddDynamicArguments("--single-branch", "--branch", opts.SingleBranch)
}
if stdout, _, err := cloneCmd.AddDynamicArguments(oldRepoPath, repoPath).
```

Reply:
Done. `"--single-branch", "--branch"` had to be added by `AddArguments` though